HopsanGUI/loadFunctions.cpp

```diff
@@ -230,6 +230,26 @@ void loadParameterValue(QDomElement &rDomElement, ModelObject* pObject, UndoStat
         return;
     }
 
+    if(existingNames.contains(parameterValue) &&
```

Comment:
The previous behavior of setParameter first tried to evaluated the parameter as if was self.parameter (inside the local component) then it would try with a system variable.

This replacement fix will correctly replace paramname with self.paramname in that case, but in the case where the paramname exist both in self and in parent it will not replace the name and we will now choose the parent version instead, which is not the same as before.  

Does this make sense ?
If so, maybe just drop the check that it is not in the parent container.
The same problem (if it is one) exists in the other place in this file with the same code.